tsdb/chunks/chunks.go

```diff
@@ -252,63 +254,131 @@ func MergeOverlappingChunks(chks []Meta) ([]Meta, error) {
 			last++
 			continue
 		}
-		nc := &newChks[last]
-		if c.MaxTime > nc.MaxTime {
-			nc.MaxTime = c.MaxTime
-		}
-		chk, err := MergeChunks(nc.Chunk, c.Chunk)
+		dividedChunkOne, dividedChunkTwo, err := MergeChunks(newChks[last].Chunk, c.Chunk, MaxSamplesInChunk)
```

Comment:
```suggestion
		chk1, chk2, err := MergeChunks(newChks[last].Chunk, c.Chunk, MaxSamplesInChunk)
```
Is fine really (:

Reply:
Also.. why two?  I would expect slice of chunks TBH.

Reply:
In this sense I don't agree with @codesome here: https://github.com/prometheus/prometheus/pull/6760#discussion_r376347819

Reply:
Also, do I need to write test cases to include the event if users change the chunk size to what they want (instead of using a fixed 120)?

Reply:
I was assuming that we will have max 120 samples in the incoming chunk. But yes, that is not necessary. But where do we expect way more than 120 samples? :)
It would be interesting to see if there are any performance implications for returning a slice. If it is very less, then it's a no brainer, else would want to consider tolerating a small spillover for the 2nd chunk.

Reply:
I was assuming that we will have max 120 samples in the incoming chunk. But yes, that is not necessary. But where do we expect way more than 120 samples? :)
It would be interesting to see if there are any performance implications for returning a slice. If it is very less, then it's a no brainer, else would want to consider tolerating a small spillover for the 2nd chunk.

Reply:
> Also, do I need to write test cases to include the event if users change the chunk size to what they want (instead of using a fixed 120)?

I don't think so. It is not user configurable.